src/postgres-util/src/lib.rs

```diff
@@ -130,6 +131,27 @@ impl AstDisplay for PgColumn {
 }
 impl_display!(PgColumn);
 
+impl TryFrom<PgColumn> for ColumnDef<Raw> {
+    type Error = anyhow::Error;
+    fn try_from(col: PgColumn) -> Result<Self, anyhow::Error> {
+        let pg_column_string = format!("({})", &col.to_ast_string_stable());
+        let parsed = parse_columns(&pg_column_string);
```

Comment:
I realize that most code was already here but... ooof the format–parse roundtrip to convert to ColumnDef is not ideal. It's pretty weird, too, that all these types are implementing `AstDisplay` when they are not actually part of the SQL parser's AST. At some point I'd love if this impl were converted to look more like `TryFrom<TableInfo> for PgTable<Raw>`, where the transformation goes from Rust type to Rust type directly.

Reply:
My thinking here was that since logic already existed to convert this type which just required using the string serialization as an intermediary then the benefits of having the logic stay in one place easily outweighed the small cost for a function which should only be called a few times during initial creation and server restart. 
I do not really know the origin story for `postgres_util` and why its types implement AstFormat; however, since all of this functionality is there and is only used outside of hot code paths I do think its the expeditious choice to use it in support of fixing the correctness bug in [MaterializeInc/materialize/issues/6739] and then we can refactor this to directly convert between types at the appropriate later date.

Reply:
I totally hear you, but rushing into changing the output of purification has gotten us into trouble in the past. It's not the efficiency that has me worried.. it's the fact that we're doing a lossy/fallible conversion here at all. These sorts of conversions have the tendency to cause bugs, from incorrect escaping resulting in bricked catalogs to type-mismatches that cause data loss.